lib/galaxy/jobs/actions/post.py

```diff
@@ -427,7 +427,7 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
                 else:
                     log.debug("No job found yet for wfi_step %s, (step %s)" % (wfi_step, wfi_step.workflow_step))
             for j2c in jobs_to_check:
-                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset.creating_job]
+                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset and x.dataset.creating_job]
```

Comment:
Should we add some logging in order to find out what are the "offending" `input_datasets`? E.g.:
```python
creating_jobs = []
for x in j2c.input_datasets:
    if not x.dataset:
        log.debug("...")
    elif not x.dataset.creating_job:
        log.debug("...")
    else:
        creating_jobs.append((x, x.dataset.creating_job))
```